js/views/userCard.js

```diff
@@ -4,19 +4,38 @@ import loadTemplate from '../utils/loadTemplate';
 import app from '../app';
 import { followedByYou, followUnfollow } from '../utils/follow';
 import Profile from '../models/profile/Profile';
+import UserCard from '../models/UserCard';
 
 export default class extends BaseVw {
   constructor(options = {}) {
     super(options);
     this.options = options;
-    this.guid = options.guid || this.model.get('guid');
+
+    if (this.model instanceof Profile) {
+      this.guid = this.model.id;
+      this.fetched = true;
+    } else if (this.model) {
+      this.guid = this.model.get('guid');
+      this.fetched = false;
+    } else {
+      this.model = new UserCard({ guid: options.guid });
```

Comment:
Since we're expecting the guid to have been passed in as an option in this block, can we please check that happened and give a descriptive error message if not? Otherwise, the profile fetch will fail because it will be trying to call `ob/profile/undefined` and it's harder for a dev to understand from that that they need to pass in a guid to this view.